app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetController.scala

```diff
@@ -63,28 +65,28 @@ class EditSetController @Inject() (
 
   private def generateEditSetView(id: String, user: User, editSetEntryRowOrder: EditSetEntryRowOrder)(implicit
     request: Request[AnyContent]
-  ): Result = {
-    val currentEditSet = editSetService.getEditSet(id)
-    val pageNumber = queryStringValue(request, offsetKey).map(_.toInt).getOrElse(1)
-    val sortedEntries = currentEditSet.entries.sorted(editSetEntryRowOrder.currentOrdering)
+  ): Future[Result] =
+    editSetService.getEditSet(id).map { currentEditSet =>
+      val pageNumber = queryStringValue(request, offsetKey).map(_.toInt).getOrElse(1)
+      val sortedEntries = currentEditSet.entries.sorted(editSetEntryRowOrder.currentOrdering)
 
-    val editSetPage = new EditSetPagination(
-      id = id,
-      rowOrder = editSetEntryRowOrder,
-      nextText = resolveMessage("edit-set.pagination.next"),
-      previousText = resolveMessage("edit-set.pagination.previous")
-    ).makeEditSetPage(sortedEntries, pageNumber)
+      val editSetPage = new EditSetPagination(
+        id = id,
+        rowOrder = editSetEntryRowOrder,
+        nextText = resolveMessage("edit-set.pagination.next"),
+        previousText = resolveMessage("edit-set.pagination.previous")
+      ).makeEditSetPage(sortedEntries, pageNumber)
 
-    val title = resolveMessage("edit-set.title", editSetPage.pageNumber, editSetPage.totalPages)
-    val heading: String = resolveMessage(
-      "edit-set.heading",
-      currentEditSet.name,
-      editSetPage.numberOfFirstEntry.toString,
-      editSetPage.numberOfLastEntry.toString,
-      editSetPage.totalNumberOfEntries.toString
-    )
-    Ok(editSet(user, title, heading, editSetPage, editSetEntryRowOrder))
-  }
+      val title = resolveMessage("edit-set.title", editSetPage.pageNumber, editSetPage.totalPages)
+      val heading: String = resolveMessage(
+        "edit-set.heading",
+        currentEditSet.name,
+        editSetPage.numberOfFirstEntry.toString,
+        editSetPage.numberOfLastEntry.toString,
+        editSetPage.totalNumberOfEntries.toString
+      )
+      Ok(editSet(user, title, heading, editSetPage, editSetEntryRowOrder))
+    }
```

Comment:
It's outside the scope, but I wonder if you could remove the unknown members in the companion object?

They are: `EditSetReorder`, `orderDirectionAscending` and `orderDirectionDescending`.

Reply:
Confirmed.